github/github.go

```diff
@@ -0,0 +1,86 @@
+package github
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/ONSdigital/dp-cli/out"
+)
+
+const (
+	LATEST_RELEASE_URL         = "https://api.github.com/repos/ONSdigital/dp-cli/releases/latest"
+	CURRENT_RELEASE_URL_PREFIX = "https://api.github.com/repos/ONSdigital/dp-cli/releases/tags"
+	DATE_TIME_FORMAT           = "2006-01-02T15:04:05Z"
+)
+
+type release struct {
+	TagName     string `json:"tag_name"`
+	PublishedAt string `json:"published_at"`
+}
+
+type rateLimit struct {
+	Message string `json:"message"`
+}
+
+func CheckApplicationVersion(version string) string {
+	version = strings.TrimSuffix(version, "-dirty")
+
+	latest := getRelease(LATEST_RELEASE_URL)
+	latestBody := getReleaseBody(latest.Body)
+	latestRelease := marshalBody(latestBody)
+
+	checkForRateLimiting(latestRelease, latestBody)
+
+	current := getRelease(fmt.Sprintf("%s/%s", CURRENT_RELEASE_URL_PREFIX, version))
+	currentBody := getReleaseBody(current.Body)
+	currentRelease := marshalBody(currentBody)
+
+	currentVersion, _ := time.Parse(DATE_TIME_FORMAT, currentRelease.PublishedAt)
+	latestVersion, _ := time.Parse(DATE_TIME_FORMAT, latestRelease.PublishedAt)
+
+	if currentVersion != latestVersion {
```

Comment:
OOI rather than comparing dates, why can't you compare `version` with the `latestRelease.TagName`?

Reply:
The TagNames are alpha numeric so the compares weren't being reliable. Just found this to be easier and more accurate at the time.